<html>
    <head>
    </head>
    <body>
        <div class="large-9 columns" id="app-details-left">
            <div>
                <h2>
                    Inspiration
                </h2>
                <p>
                    We wanted our hack to combine some of the domains we are interested in and to solve a small part of an important problem. We've decided to focus on an issue in bioinformatics &ndash; handling the data from next generation sequencing.
                </p>
                <h2>
                    What it does
                </h2>
                <p>
                    We were interested in separation of sequences from biological noise and in data clustering. Traditionally, the methods that are used to solve these problems are heavily parameterized &ndash; for example, principal component analysis and hierarchical clustering &ndash; making the clustering less robust and requiring precise parameter setup. Our hack consists of a data analysis and visualization package in Python.
                </p>
                <p>
                    <img alt="Data" data-canonical-url="http://i.imgur.com/y3E4T4O.png" src="https://res.cloudinary.com/devpost/image/fetch/s--aKY3Lgt1--/c_limit,f_auto,fl_lossy,q_auto:eco,w_900/http://i.imgur.com/y3E4T4O.png">
                </p>
                <p>
                    The analyzed data were taken from real experiments that look at the mutations occurring during the cycles of evolution of RNA molecules. These are common types of experiments in such fields like virology and origins of life.
                </p>
                <h2>
                    How we built it
                </h2>
                <p>
                    We've developed a method that is based on the graph theory approaches. We represent the sequences as nodes of a graph, where edges and their weight capture how similar these sequences are.
                </p>
                <p>
                    <img alt="Initial graph" data-canonical-url="http://i.imgur.com/bwT7lT9.png" src="https://res.cloudinary.com/devpost/image/fetch/s--C2tpcMuo--/c_limit,f_auto,fl_lossy,q_auto:eco,w_900/http://i.imgur.com/bwT7lT9.png">
                </p>
                <p>
                    Such representation allows us to use the powerful network science toolbox for further analysis. The initial graph is extremely dense &ndash; almost every node is connected to every other. We've removed the edge with insufficient edge weight.
                </p>
                <p>
                    <img alt="Simplified graph" data-canonical-url="http://i.imgur.com/H3FC1fb.png" src="https://res.cloudinary.com/devpost/image/fetch/s--gKx4Jcsy--/c_limit,f_auto,fl_lossy,q_auto:eco,w_900/http://i.imgur.com/H3FC1fb.png">
                </p>
                <p>
                    We have used the Louvain Method to discover the underlying community structure of the graph. The method divides the nodes into clusters, maximizing their internal connectivity and minimizing the external connections between them. Such approach is a good fit to our problem, as the related sequences have a high similarity index, translating into high edge weight and, thus, high internal connectivity. The resulting community structure divides the sequences according to their similarities. The method does not require any precise parameter setup and is parallel, making it appropriate for large-scale experiments (we were able to process a graph with 2 million edges on a regular laptop).
                </p>
                <p>
                    <img alt="Community structure" data-canonical-url="http://i.imgur.com/cWijoa8.png" src="https://res.cloudinary.com/devpost/image/fetch/s--SX3UqA9N--/c_limit,f_auto,fl_lossy,q_auto:eco,w_900/http://i.imgur.com/cWijoa8.png">
                </p>
                <p>
                    We used Python for implementing the method. A number of libraries was used: NumPy, NetworkX, NetworKit, Matplotlib, IPython, and others.
                </p>
                <h2>
                    Challenges we ran into
                </h2>
                <p>
                    The time required to process large datasets was outside the scope of the event (sequence alignment is very time consuming), thus we had to use medium-sized datasets.
                </p>
                <h2>
                    Accomplishments that we proud of
                </h2>
                <p>
                    Using experimental RNA data, we were able to demonstrate that such method does achieve its goal &ndash; it robustly separates the sequences with the similar mutation profiles.  We've tried the method on 7 sets of data. In every case, it yielded satisfactory results.
                </p>
                <h2>
                    What we learned
                </h2>
                <p>
                    We've gained additional knowledge about bioinformatics and scientific computing.
                </p>
                <h2>
                    What's next for BClust
                </h2>
                <p>
                    Other network science techniques could be applied to the sequence graphs. We have already considered link prediction and dynamic graph partitioning. We are hoping to validate our technique on a larger dataset.
                </p>
            </div>
            <div class="" id="built-with">
                <h2>
                    Built With
                </h2>
                <ul class="no-bullet inline-list">
                    <li>
                        <span class="cp-tag">
                            matplotlib
                        </span>
                    </li>
                    <li>
                        <span class="cp-tag">
                            networkit
                        </span>
                    </li>
                    <li>
                        <span class="cp-tag">
                            networkx
                        </span>
                    </li>
                    <li>
                        <span class="cp-tag recognized-tag">
                            <a href="https://devpost.com/software/built-with/numpy">
                                numpy
                            </a>
                        </span>
                    </li>
                    <li>
                        <span class="cp-tag recognized-tag">
                            <a href="https://devpost.com/software/built-with/python">
                                python
                            </a>
                        </span>
                    </li>
                </ul>
            </div>
        </div>
    </body>
</html>
