<html>
    <head>
    </head>
    <body>
        <div class="large-9 columns" id="app-details-left">
            <div id="gallery">
                <ul>
                    <li class="text-center">
                        <a data-lightbox="652603" data-title="Full causal network of CYP450 mutations showing delineating treks representing subpopulation development" href="https://challengepost-s3-challengepost.netdna-ssl.com/photos/production/software_photos/000/652/603/datas/original.png">
                            <img alt="CYP450 Mutations Phylogenetic Tree using Causal Model &ndash; screenshot 1" class="software_photo_image image-replacement" onerror="this.onerror=null;this.src='https://devpost-challengepost.netdna-ssl.com/assets/defaults/thumbnail-placeholder-42bcab8d8178b413922ae2877d8b0868.gif';" src="//challengepost-s3-challengepost.netdna-ssl.com/photos/production/software_photos/000/652/603/datas/gallery.jpg">
                        </a>
                        <span class="expand-tag">
                            <i class="fas fa-expand">
                            </i>
                        </span>
                        <p>
                            <i>
                                Full causal network of CYP450 mutations showing delineating treks representing subpopulation development
                            </i>
                        </p>
                    </li>
                    <li class="text-center">
                        <a data-lightbox="652604" data-title="causal network of CYP450 mutations zoomed in to most commonly shared variants among populations" href="https://challengepost-s3-challengepost.netdna-ssl.com/photos/production/software_photos/000/652/604/datas/original.png">
                            <img alt="CYP450 Mutations Phylogenetic Tree using Causal Model &ndash; screenshot 2" class="software_photo_image image-replacement" onerror="this.onerror=null;this.src='https://devpost-challengepost.netdna-ssl.com/assets/defaults/thumbnail-placeholder-42bcab8d8178b413922ae2877d8b0868.gif';" src="//challengepost-s3-challengepost.netdna-ssl.com/photos/production/software_photos/000/652/604/datas/gallery.jpg">
                        </a>
                        <span class="expand-tag">
                            <i class="fas fa-expand">
                            </i>
                        </span>
                        <p>
                            <i>
                                causal network of CYP450 mutations zoomed in to most commonly shared variants among populations
                            </i>
                        </p>
                    </li>
                    <li class="text-center">
                        <a data-lightbox="652608" data-title="Linkage disequilibrum plot in major subpopulations demonstrates that more linkage is observable when we cluster" href="https://challengepost-s3-challengepost.netdna-ssl.com/photos/production/software_photos/000/652/608/datas/original.png">
                            <img alt="CYP450 Mutations Phylogenetic Tree using Causal Model &ndash; screenshot 3" class="software_photo_image image-replacement" onerror="this.onerror=null;this.src='https://devpost-challengepost.netdna-ssl.com/assets/defaults/thumbnail-placeholder-42bcab8d8178b413922ae2877d8b0868.gif';" src="//challengepost-s3-challengepost.netdna-ssl.com/photos/production/software_photos/000/652/608/datas/gallery.jpg">
                        </a>
                        <span class="expand-tag">
                            <i class="fas fa-expand">
                            </i>
                        </span>
                        <p>
                            <i>
                                Linkage disequilibrum plot in major subpopulations demonstrates that more linkage is observable when we cluster
                            </i>
                        </p>
                    </li>
                    <li class="text-center">
                        <a data-lightbox="652609" data-title="Linkage disequilibrium is entirely hidden when we observe all participant data together without clustering first" href="https://challengepost-s3-challengepost.netdna-ssl.com/photos/production/software_photos/000/652/609/datas/original.png">
                            <img alt="CYP450 Mutations Phylogenetic Tree using Causal Model &ndash; screenshot 4" class="software_photo_image image-replacement" onerror="this.onerror=null;this.src='https://devpost-challengepost.netdna-ssl.com/assets/defaults/thumbnail-placeholder-42bcab8d8178b413922ae2877d8b0868.gif';" src="//challengepost-s3-challengepost.netdna-ssl.com/photos/production/software_photos/000/652/609/datas/gallery.jpg">
                        </a>
                        <span class="expand-tag">
                            <i class="fas fa-expand">
                            </i>
                        </span>
                        <p>
                            <i>
                                Linkage disequilibrium is entirely hidden when we observe all participant data together without clustering first
                            </i>
                        </p>
                    </li>
                    <li class="text-center">
                        <a data-lightbox="652610" data-title="Linkage disequilibrium presents expected associations when we look at each ethnic groups separately" href="https://challengepost-s3-challengepost.netdna-ssl.com/photos/production/software_photos/000/652/610/datas/original.png">
                            <img alt="CYP450 Mutations Phylogenetic Tree using Causal Model &ndash; screenshot 5" class="software_photo_image image-replacement" onerror="this.onerror=null;this.src='https://devpost-challengepost.netdna-ssl.com/assets/defaults/thumbnail-placeholder-42bcab8d8178b413922ae2877d8b0868.gif';" src="//challengepost-s3-challengepost.netdna-ssl.com/photos/production/software_photos/000/652/610/datas/gallery.jpg">
                        </a>
                        <span class="expand-tag">
                            <i class="fas fa-expand">
                            </i>
                        </span>
                        <p>
                            <i>
                                Linkage disequilibrium presents expected associations when we look at each ethnic groups separately
                            </i>
                        </p>
                    </li>
                </ul>
            </div>
            <div>
                <h2>
                    Inspiration
                </h2>
                <p>
                    Determine whether causal discovery methods can be implemented in population genetics produce phylogenetic trees of mutations
Currently, this is done with stochastic modeling techniques (i.e., Markov Chain Monte Carlo), which are very computationally heavy.
                </p>
                <p>
                    CYP450: super-family of over 50 proteins that are involved in the metabolism of many drugs; just six of these proteins metabolize 90% of drugs. Understanding ethnic differences in drug response is an important area in pharmacogenomics.
                </p>
                <h2>
                    What it does
                </h2>
                <p>
                    The information produced can be useful for precision medicine hypothesis testing. For example, it can assist in moving away from the "one SNP, one disease" association hypotheses and towards a more comprehensive notion that sets of mutations together contribute to disease. In the latter case, it is important to consider ethnicity (or relatedness) of individuals in order to account for the proper linkage and progression of mutations. In the causal plot, we can observe the direction (or order) in mutation acquisition and how the separation of populations into subgroups leads to isolated chains of mutation development.
                </p>
                <h2>
                    How I built
                </h2>
                <p>
                    The dataset was obtained from the 1000 Genomes Project, which contains whole genome sequencing of 2,504 individuals belonging to one of 26 different ethnic groups. I accessed this publicly available data through 1000 Genomes Browser and downloaded it as VCF files containing SNPs for the entire CYP450 locus (7,274 base pairs).
                </p>
                <p>
                    I used R for data curation and conversion of the data format into TSV.
Tetrad GUI (v6.5.3) was used for model determination and computation using FGES algorithm with discrete BIC scoring. Cytoscape for visualizing directed mutation causation.
                </p>
                <h2>
                    Challenges I ran into
                </h2>
                <p>
                    Variant calling files (VCF) are quite complicated in structure, so it was challenging to parse it out into a data frame that is readable in R.
                </p>
                <h2>
                    Accomplishments that I'm proud of
                </h2>
                <p>
                    I am proud of the fact that I was able to take the approaches learned in the Causal Discovery Short Course to find a more robust means of producing ancestral phylogenetic trees of mutation than traditional stochastic modeling.
                </p>
                <h2>
                    What I learned
                </h2>
                <p>
                    I learned the theoretical and computational approach for causal discovery using directed paths and treks for modeling causal relatedness of variables, implementing algorithms that determine a Bayesian Network. Additionally, I learned to do this using the Tetrad graphical user interface software developed by the Center for Causal Discovery.
                </p>
                <h2>
                    What's next for CYP450 Mutations Phylogenetic Tree using Causal Model
                </h2>
                <p>
                    The next step of this project is to annotate the phylogenetic tree with ethnic information. This can be done in many ways, but first I will try annotating each SNP (node) with the ethnic group that has the highest prevalence of the mutation. 
Additionally, I will seek out a way to annotate entire treks so that I can look at the delineating treks and find out which ethnic group best matches this divergence based on their genotypes.
                </p>
            </div>
            <div class="" id="built-with">
                <h2>
                    Built With
                </h2>
                <ul class="no-bullet inline-list">
                    <li>
                        <span class="cp-tag">
                            cytoscape
                        </span>
                    </li>
                    <li>
                        <span class="cp-tag recognized-tag">
                            <a href="https://devpost.com/software/built-with/r">
                                r
                            </a>
                        </span>
                    </li>
                    <li>
                        <span class="cp-tag">
                            tetrad-gui
                        </span>
                    </li>
                </ul>
            </div>
        </div>
    </body>
</html>
