<html>
    <head>
    </head>
    <body>
        <div class="large-9 columns" id="app-details-left">
            <div id="gallery">
                <ul>
                    <li class="text-center">
                        <a data-lightbox="368298" data-title="beast lab computer running my script, using 44 threads at once!" href="https://challengepost-s3-challengepost.netdna-ssl.com/photos/production/software_photos/000/368/298/datas/original.png">
                            <img alt="metagenome annotation script &ndash; screenshot 1" class="software_photo_image image-replacement" onerror="this.onerror=null;this.src='https://devpost-challengepost.netdna-ssl.com/assets/defaults/thumbnail-placeholder-42bcab8d8178b413922ae2877d8b0868.gif';" src="//challengepost-s3-challengepost.netdna-ssl.com/photos/production/software_photos/000/368/298/datas/gallery.jpg">
                        </a>
                        <span class="expand-tag">
                            <i class="fas fa-expand">
                            </i>
                        </span>
                        <p>
                            <i>
                                beast lab computer running my script, using 44 threads at once!
                            </i>
                        </p>
                    </li>
                </ul>
            </div>
            <div>
                <h2>
                    Inspiration
                </h2>
                <p>
                    I am studying the human gut microbiome by sequencing all the bacterial DNA in people's poop. One of the experiments I'm running involves figuring out what proteins all the DNA code for, and seeing if any protein DNA sequences are relatively more abundant in the gut microbiomes of healthy people compared to the gut microbiome of people with disease. I previously built a reference library of bacterial sequences annotated with protein functions. Because most bacteria have not been sequenced, I was only able to determine what protein were encoded in the DNA for a third of my sequences. To figure out what proteins were encoded by my other sequences, I wanted to assemble my DNA into partial genomes, and query them with the SEED protein annotation database (
                    <a href="http://www.theseed.org/wiki/Main_Page" rel="nofollow">
                        www.theseed.org/wiki/Main_Page
                    </a>
                    ).
                </p>
                <h2>
                    What it does
                </h2>
                <p>
                    The code takes sequences which have already been assembled by Trinity (
                    <a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki" rel="nofollow">
                        https://github.com/trinityrnaseq/trinityrnaseq/wiki
                    </a>
                    ), and uses the BLAST algorithm (through the command line tool
                    <a href="http://www.ncbi.nlm.nih.gov/books/NBK279690/" rel="nofollow">
                        http://www.ncbi.nlm.nih.gov/books/NBK279690/
                    </a>
                    ) to compare the sequences with the SEED protein annotation database.
                </p>
                <h2>
                    How we built it
                </h2>
                <p>
                    BLAST is run on the command line through bash scripts, and the amalgamation and extraction of sequence data is done with Perl. Bash scripts were also used to run the annotation individually for each microbiome sample.
                </p>
                <h2>
                    Challenges we ran into
                </h2>
                <ul>
                    <li>
                        Some of the sequences are quite long, but it is possible for the first hundreds of hits in the SEED database to annotate the same sequence. To account for this, I took the best non-overlapping protein matches, and if any of the unmatched sequence segments were 500 nucleotides or longer, I queried them against the SEED database again.
                    </li>
                    <li>
                        I spent an inordinate amount of time debugging my code that checked if the positions of two protein matches overlapped. Flashback to my first mock interview, where I spent 45 minutes blanking on how to determine if two rectangles overlap. D:
                    </li>
                </ul>
                <h2>
                    Accomplishments that we're proud of
                </h2>
                <p>
                    The code works!
                </p>
                <h2>
                    What we learned
                </h2>
                <ul>
                    <li>
                        In Perl, use "if ((x&gt;0 &amp;&amp; x&lt;10) || (y&gt;0 &amp;&amp; y&lt;10))" rather than 'if ((x&gt;0 and x&lt;10) or (y&gt;0 and y&lt;10))".
                    </li>
                    <li>
                        If you are checking to see if two things overlap, don't forget to check the case where one thing is completely inside the other thing
                    </li>
                </ul>
                <h2>
                    What's next for metagenome annotation script
                </h2>
                <p>
                    The BLAST is probably going to take a week to run. I need it to finish faster so I can write it up in my thesis!
                </p>
                <p>
                    Note: The relevant part of the GitHub repository linked is under the heading "Create fasta file for bowtie index".
                </p>
            </div>
            <div class="" id="built-with">
                <h2>
                    Built With
                </h2>
                <ul class="no-bullet inline-list">
                    <li>
                        <span class="cp-tag recognized-tag">
                            <a href="https://devpost.com/software/built-with/bash">
                                bash
                            </a>
                        </span>
                    </li>
                    <li>
                        <span class="cp-tag">
                            blast
                        </span>
                    </li>
                    <li>
                        <span class="cp-tag recognized-tag">
                            <a href="https://devpost.com/software/built-with/perl">
                                perl
                            </a>
                        </span>
                    </li>
                </ul>
            </div>
            <nav class="app-links section">
                <h2>
                    Try it out
                </h2>
                <ul class="no-bullet" data-role="software-urls">
                    <li>
                        <a href="https://github.com/ruthgrace/exploring_nafld_assembly" rel="nofollow" target="_blank" title="https://github.com/ruthgrace/exploring_nafld_assembly">
                            <i class="ss-icon ss-link">
                            </i>
                            <span>
                                github.com
                            </span>
                        </a>
                    </li>
                </ul>
            </nav>
        </div>
    </body>
</html>
